test/module.test.js

```diff
@@ -201,6 +201,58 @@ describe('Test sidekick', () => {
     assert.ok(plugins.find((p) => p.id === 'foo'), 'Did not add plugin from config');
   }).timeout(IT_DEFAULT_TIMEOUT);
 
+  it('Adds badge plugin to feature container', async () => {
+    const setup = new Setup('blog');
+    nock.sidekick(setup, {
+      configJson: `{
+        "plugins": [{
+          "id": "foo",
+          "title": "Foo",
+          "isBadge": true
+        },
+        {
+          "id": "foo2",
+          "title": "Foo 2",
+          "isBadge": true,
+          "badgeTextColor": "red",
+          "badgeBackgroundColor": "#fff"
+        },
+        {
+          "id": "foo3",
+          "title": "Foo 3",
+          "isBadge": true,
+          "badgeTextColor": "#abc",
+          "badgeBackgroundColor": "#def",
+          "badgeTextColorDark": "#000",
+          "badgeBackgroundColorDark": "#fff"
+        }]
+      }`,
+    });
+    nock.admin(setup);
+    const { checkPageResult, plugins } = await new SidekickTest({
+      browser,
+      page,
+      loadModule,
+      checkPage: (p) => p.evaluate(() => {
+        const sk = window.hlx.sidekick;
+        let result = true;
+        // test default badge
+        let badge = sk.get('foo').querySelector('span');
+        result = result && badge.closest('div.feature-container') && getComputedStyle(badge).color === 'rgb(80, 80, 80)' && getComputedStyle(badge).backgroundColor === 'rgb(192, 192, 192)';
+        // test badge with custom colors
+        badge = sk.get('foo2').querySelector('span');
+        result = result && badge.closest('div.feature-container') && getComputedStyle(badge).color === 'rgb(255, 0, 0)' && getComputedStyle(badge).backgroundColor === 'rgb(255, 255, 255)';
+        // test badge with custom colors and dark mode
+        badge = sk.get('foo3').querySelector('span');
+        result = result && badge.closest('div.feature-container') && getComputedStyle(badge).color === 'rgb(170, 187, 204)' && getComputedStyle(badge).backgroundColor === 'rgb(221, 238, 255)';
+        return result;
```

Comment:
Shouldn't you return an array of 3 results? Otherwise the test is considered successful even if the first two checks are failing.